astropy/convolution/convolve.py

```diff
@@ -190,6 +190,9 @@ def convolve(array, kernel, boundary='fill', fill_value=0.,
     passed_kernel = kernel
     passed_array = array
 
+    # Get array quantity if exists
+    passed_quantity = getattr(passed_array, "unit", None)
```

Comment:
Nitpicky, but I think it would be clearer in this variable is called `array_unit` (or something similar)